pandas/tests/reshape/test_pivot.py

```diff
@@ -286,6 +286,27 @@ def test_pivot_with_interval_index(self, interval_values, dropna):
         expected = DataFrame({"B": 1}, index=Index(interval_values.unique(), name="A"))
         tm.assert_frame_equal(result, expected)
 
+    def test_pivot_with_interval_index_margins(self):
+        # GH 25815
+        ordered_cat = pd.IntervalIndex.from_arrays([0, 0, 1, 1], [1, 1, 2, 2])
+        df = pd.DataFrame(
+            {
+                "A": np.arange(4, 0, -1).astype('int32'),
+                "B": ["a", "b", "a", "b"],
+                "C": pd.Categorical(ordered_cat, ordered=True).sort_values(
+                    ascending=False
+                ),
+            }
+        )
+
+        pivot_tab = pd.pivot_table(
+            df, index="C", columns="B", values="A", aggfunc="sum", margins=True
+        )
+
+        result = pivot_tab["All"]
+        expected = pd.Series([3, 7, 10], index=result.index, name="All", dtype="int32")
```

Comment:
```suggestion
        expected = Series(
            [3, 7, 10],
            index=Index([pd.Interval(0, 1), pd.Interval(1, 2), "All"], name="C"),
            name="All",
        )
```

Reply:
Why not using index from result , seems easier ?

Reply:
I'm not sure that would be in the spirit of constructing the expected result explicitly as suggested in https://github.com/pandas-dev/pandas/pull/27245#discussion_r300725193